extensions/kafka-client/deployment/src/main/java/io/quarkus/kafka/client/deployment/SnappyUtils.java

```diff
@@ -0,0 +1,22 @@
+package io.quarkus.kafka.client.deployment;
+
+import org.xerial.snappy.OSInfo;
+
+/**
+ * This class should only be used if Snappy is available on the classpath.
+ */
+public class SnappyUtils {
+
+    private SnappyUtils() {
+        // Avoid direct instantiation
+    }
+
+    public static String getNativeLibFolderPathForCurrentOS() {
+        try {
+            return OSInfo.getNativeLibFolderPathForCurrentOS();
+        } catch (Exception e) {
+            e.printStackTrace();
+            return null;
```

Comment:
I don't know what your team policy on the Clean Code "never pass null, never return null"-mantra is but to me this is a code smell. How about returning an `Optional` instead?

Reply:
Actually it's a left-over, it should not be there.